test/gurka/test_build_admin.cc

```diff
@@ -27,76 +25,129 @@ valhalla::gurka::map BuildPBF(const std::string& workdir) {
         |       |       |   |   |       |
         |       |       |   Z   |       |
         F-------E-------D-------L-------K
+
+
+
+        M-------N-------O
+        |    S-----T    |
+        |    |     |    |
+        |  a-|--b--|-c  |
+        |    |     |    |
+        |    U-----V    |
+        Q-------R-------P
+
   )";
 
   // To define an administrative boundary, the nodes must form a closed polygon.
-  const gurka::ways ways = {{"ABCDEFA", {}},
-                            {"ABEFA", {}},
-                            {"BCDEB", {}},
-                            {"CIJKLDC", {}},
-                            {"CILDC", {}},
-                            {"IJKLI", {}},
-                            {"GH",
-                             {
-                                 {"highway", "primary"},
-                             }},
-                            {"HX",
-                             {
-                                 {"highway", "primary"},
-                             }},
-                            {"XY",
-                             {
-                                 {"highway", "primary"},
-                             }},
-                            {"WX",
-                             {
-                                 {"highway", "primary"},
-                             }},
-                            {"XZ",
-                             {
-                                 {"highway", "primary"},
-                             }}};
-
-  // X lives Japan which allows named intersections - and is named.
+  const gurka::ways ways = {
+      {"AB", {}},
+      {"AF", {}},
+      {"EF", {}},
+      {"EB", {}},
+      {"BCDE", {}},
+      {"CIJKLDC", {}},
+      {"CILDC", {}},
+      {"IJKLI", {}},
+      {"MNOPRQM", {}},
+      {"STVUS", {}},
+      {"GH",
```

Comment:
a lot of the ways/relations/etc has diffs because of formatting but i did change some of them from relations with a single member to be multiple members with different directions to see if the geometry would still be pieced together properly.